feat(android): document how to use renderdoc with Android (#33)
canonical/anbox-cloud-docs: howto/android/graphics-debugging-with-renderdoc.md

[Renderdoc](https://github.com/baldurk/renderdoc) is a graphics frame debugger which supports Android applications.

> Debugging graphics can be challenging and [Renderdoc](https://github.com/baldurk/renderdoc) is a great tool to provide insight into Vulkan and OpenGL command execution. The following describes how you can use Renderdoc with Anbox Cloud.

Reviewer: Isn't the first sentence redundant with the second one?

Author: Yes, I will drop the first one.

## Install Renderdoc

> Renderdoc provides binary builds [here](https://renderdoc.org/builds). These come with all prerequisites to be able to capture on Android.

Author: I can rework that, thanks!

> For capturing Android applications you also need the Android SDK as Renderdoc needs a few tools like adb or aapt. The easiest is to install [Android Studio](https://developer.android.com/studio) which will allow you to install the SDK.

Reviewer: Shouldn't we instead tell people to install from the snap here? We could link to the snapstore https://snapcraft.io/android-studio, or use the CLI to install:
```sh
sudo snap install android-studio --classic
```

What do you think?

Author: No, we should only ever point to the official installation instructions of external tools we point to and do not reference custom packaged, not official ways of installing.

## Configure Renderdoc

Capturing applications running in Android with Renderdoc is not much different than what is described in [the official documentation](https://renderdoc.org/docs/how/how_android_capture.html). The most important part is to configure the path to the Android SDK in the Renderdoc settings. See [here](https://renderdoc.org/docs/window/settings_window.html#android-options) for more details.

## Configure Anbox

In order to run Renderdoc with Anbox Cloud you need to create an instance with its ADB port exposed.

    amc launch -s +adb ...

Afterwards you connect to Android by running

    adb connect <host IP>:9001

The device will show up in the output of

    adb devices

> Renderdoc can only inject itself into applications which have debug mode turn on. See [here](https://renderdoc.org/docs/how/how_android_capture.html#how-do-i-use-renderdoc-on-android) for more details. To be able to debug any application regardless of it being build with debug settings Renderdoc needs to be able to run as root over ADB. For this the `android.enable_adb_root` feature flag can be set to allow this. Afterwards can run Renderdoc with any application.

Reviewer: ```suggestion
Renderdoc can only inject itself into applications which have debug mode turn on. See [here](https://renderdoc.org/docs/how/how_android_capture.html#how-do-i-use-renderdoc-on-android) for more details. To be able to debug any application regardless of it being build with debug settings Renderdoc needs to be able to run as root over ADB. For this the `android.enable_adb_root` feature flag can be set to allow this. Afterwards you can run Renderdoc with any application.
```

Should we link to the feature flag section of our docs here?

Debugging graphics can be challenging and [Renderdoc](https://github.com/baldurk/renderdoc) is a great tool to provide insight into Vulkan and OpenGL command execution. The following describes how you can use Renderdoc with Anbox Cloud.

## Install Renderdoc

Renderdoc provides binary builds [here](https://renderdoc.org/builds). These come with all prerequisites to be able to capture on Android.

[note type="caution" status="Warning"]

> The offical Renderdoc builds only support Arm and not x86 Android builds. If you need x86 support you will have to build Renderdoc yourself from source. Please have a look at the official [repository](https://github.com/baldurk/renderdoc) for more information on how to do that.

Reviewer 2: ```suggestion
The offical Renderdoc builds only support ARM and not x86 Android builds. If you need x86 support, you will have to build Renderdoc yourself from source. The official [repository](https://github.com/baldurk/renderdoc) for Renderdoc provides more information on how to do that.
```
